src/Spectrometer-OceanOptics_OceanDirect/main.py

```diff
@@ -0,0 +1,325 @@
+            self.spectrometer = self.od.from_serial_number(serial_number)
+            # self.spectrometer.open_device()
+            self.store_parameter(self.unique_spectrometer_identifier, self.spectrometer)
+
+    def disconnect(self):
+        pass
+        
+    def initialize(self):
+
+        # Print/Logs some details about the spectrometer
+        # self.spectrometer.details()
+
+        self.dark_pixel_indices = self.spectrometer.get_electric_dark_pixel_indices()
+        # print("dark: ", self.dark_pixel_indices)
+
+        self.integration_time_min = self.spectrometer.get_minimum_integration_time()
+        self.integration_time_max = self.spectrometer.get_maximum_integration_time()
+
+        self.acquisition_delay_min = self.spectrometer.get_acquisition_delay_minimum()
+        self.acquisition_delay_max = self.spectrometer.get_acquisition_delay_maximum()
+
+        # Integration time
+        integration_time = float(self.integration_time)
+        self.set_integration_time(integration_time)
+
+        # Averages
+        self.average = int(self.average)
+        self.spectrometer.set_scans_to_average(self.average)
+
+        # Trigger
+        index = self.trigger_modes[self.trigger_mode]
+        self.spectrometer.set_trigger_mode(index)
+
+        # Acquisition delay
+        self.trigger_delay = float(self.trigger_delay)
+        self.set_acquisition_delay(self.trigger_delay)
+
+        # Electric dark correction
+        # self.spectrometer.set_electric_dark_correction_usage(False)
+
+        # Wavelengths
+        self.wavelengths = self.get_wavelengths()
+
+        # Remains for later in case calibration will be added
+        """
+        if self.calibration != "" and self.calibration != "None":
+            calibration_file = self.calibrationfolder + os.sep + self.calibration
+
+            if not calibration_file.endswith(".IrradCal"):
+                calibration_file += ".IrradCal"
+            
+            IrradCal=np.loadtxt(calibration_file, skiprows=9)  # Ocean Optics file
+            self.calibration_array = IrradCal[:,1]
+        else:
+            self.calibration_array = np.ones(self.spectrometer.pixels)
+       
+        if self.spectrometer.pixels != len(self.calibration_array):
+            self.stopMeasurement = "Check your calibration file. Number of pixels is not equal to the pixels your spectrometer has."
+        """
+
+    def apply(self):
+    
+        if self.sweep_mode.startswith("Integration time"):
+
+            integration_time = float(self.value)
+            self.set_integration_time(integration_time)
+
+    def measure(self):
+
+        self.intensities = self.get_intensities()
+
+    def call(self):
+
+        return [self.wavelengths, self.intensities, self.integration_time]
+
+
+    def check_oceandirect_api(self):
+
+        if not api_path_defined:
+            msg = "Path to OceanDirect API not found. Please install OceanDirect API before using the driver."
```

Comment:
A bit confused about the wording.
The API is an interface, something like a definition or a protocol how to communicate. But you can't install a definition. Usually what you install is an SDK or something like that. How does OceanDirect call the software themselves? The only thing I've found in the manual is they call it "OceanDirect Software" and default installation directory is "OceanDirect SDK".

Reply:
changed it.